Basic Contribution Guide (#2)
this-is-squad-goals/dotfiles: README.md

Participants asked to stop any harassing behavior are expected to comply immediately.

If a participant engages in harassing behavior, the admins may take any action they deem appropriate, up to and including expulsion from this slack and identifying the participant publicly as someone about whom we've received complaints.

**Please keep what's said in Squad Goals confidential**. Don't repeat or quote things said here without the affirmative consent of the speaker(s). When quoting (with consent), please be careful not to reveal the existence of Squad Goals. Rather, you can refer to the quote as something that was said "in chat" or while you were talking to the quoted member.

**Please be mindful that things you say here may at some point become public**. While we expect members to honor the confidentiality of this space, we cannot guarantee that they will do so--nor can we guarantee that every member's login credentials and logged-in devices are secure. Please exercise caution and refrain from sharing sensitive information that could harm you or others if it became public.

## Trello

> We keep track of what needs to be done, what others are working on, and what is already complete on our Trello board. This can be viewed by requesting an invitation in the Slack channel.

Reviewer: For this project, the goal of using a tool like Trello is mostly to keep track of what's in progress so that we don't accidentally duplicate our efforts. It also helps to have visibility into how much work any given person is taking on, because I'd rather have small incremental updates than big periodic code drops.
```suggestion
We keep track of what needs to be done, what others are working on, and what is already complete on our Trello board. This can be viewed by requesting an invitation in the Slack channel. We use Trello to keep track of things that are currently in progress, and to get a sense of how much work any given person has taken on. Please check on it periodically to be sure that your tickets are accurate and reflect the work you're doing.
```

# General Organization

We are operating on a microservice architecture. We hope that this style will allow us to remain:

-   Highly maintainable and testable
-   Loosely coupled
-   Independently deployable
-   Organized around business capabilities.

Therefore, if you want to use a certain technology or style for a portion of the project, we want this app to be a place to do so. 

Communication is a high priority for us. Please stay involved in the planning of this application.

Credit: https://microservices.io/

# Cloning and Starting a Repo

Each Repo will be different. This is a generalized guide to getting started.

1. Copy the project URL
2. In GitBash or Terminal:

```
$ git clone [project URL]
$ cd [project name]
```
3. Inside of the root directory for the project: 
```
$ npm run serve
```

# Create a Feature Branch

> We prefer to contribute and keep code organized using feature branches.

Reviewer: ```suggestion
We prefer to contribute and keep code organized using feature branches. For convenience, you may get admin access on one or more repos. Even in this case, please do not commit directly to the master branch. We use Pull Requests from a feature branch to facilitate code review within the group. Our goal with code review is to give an opportunity for two-way learning, and to discuss the design and technical decisions that were made to ensure they're consistent with the rest of the application and that they're generally understood by the team.
```

**Use the following standards to name your branch:**

> 1.HELP HERE 

Reviewer: Feature branches should not be named after the author. Feature branches should be named after the Trello card that it will address. If you're doing work that doesn't have a Trello card, please talk with the team first. This is to ensure that it's not accidentally duplicating something that already exists, and to keep people informed about the features, design, and architecture of the whole system moving forward.

> 1.HELP HERE 

Reviewer: ```suggestion
1. Feature branches should not be named after the author.
2. Feature branches should be named after the Trello card that it will address.

If you're doing work that doesn't have a Trello card, please talk with the team first. This is to ensure that it's not accidentally duplicating something that already exists, and to keep people informed about the features, design, and architecture of the whole system moving forward.
```

Before creating a new branch, pull the changes from upstream. Your master needs to be up to date.
```
$ git pull
```
If you want create a new branch:
```
$ git branch <name_of_your_new_branch>
```
Switch to this branch :
```
$ git checkout -b [name_of_your_new_branch]
```

Push the branch on github :
```
$ git push origin [name_of_your_new_branch]
```

When you want to commit something in your branch, be sure to be in your branch. Add -u parameter to set upstream.

You can see all branches created by using :
```
$ git branch -a
```

For credit and more: https://github.com/Kunena/Kunena-Forum/wiki/Create-a-new-branch-with-git-and-manage-branches

# Coding Standards

Before a merge, each pull request must contain:

> [WHAT STUFF]

Reviewer: Good question

Use a clear and descriptive commit message about what feature/bug you are working on in your commit message. Try to focus the message on the *Why* and not the *How*.

# Submit a Pull Request

To prevent bad habits, allow these protections in GitHub: 

Settings > Branches > Add Rule

Branch Name Pattern: Master
Check "Require pull request reviews before merging".

In GitBash or Terminal:
```
$ git checkout <your-feature-branch>
$ git add [files you've changed]
$ git push
```

> In GitHub in the Squad Goals project, open a pull request. This will be reviewed by another contributor. You will be alerted when this is merged into the project.

Reviewer: I'd rather not have a core group of gate keepers that are responsible for doing all review and merge. As the project grows, we might need to move to that kind of model, but for now since we're only working with people we've specifically invited, I'd like to just trust them to be considerate.

Also, it's source control, if we need to revert something, we can. Although with that in mind, it's probably best to re-base and squash before merging, so that reverts will be simpler.

```suggestion
In GitHub in the Squad Goals project, open a pull request. This will be reviewed other contributors. Please allow time for anyone who is interested to review the changes before they are merged.
```
